internal/pkg/cli/pipeline_update.go

```diff
@@ -0,0 +1,236 @@
+	if err != nil {
+		return nil, err
+	}
+
+	buckets := []deploy.ArtifactBucket{}
+	for _, resource := range regionalResources {
+		bucket := deploy.ArtifactBucket{
+			BucketName: resource.S3Bucket,
+			KeyArn: resource.KMSKeyARN,
+		}
+		buckets = append(buckets, bucket)
+	}
+
+	return buckets, nil
+}
+
+func (opts *UpdatePipelineOpts) Execute() error {
+
+	// read pipeline manifest
+	data, err := opts.ws.ReadFile(defaultPipelineFilename)
+	if err != nil {
+		return nil
+	}
+
+	pipeline, err := manifest.UnmarshalPipeline(data)
+	// convert source TODO combine?
+	// TODO nil check source?
+	source := &deploy.Source{
+		ProviderName: pipeline.Source.ProviderName,
+		Properties: pipeline.Source.Properties,
+	}
+	// convert stages to deployment stages
+	stages, err := opts.convertStages(pipeline.Stages)
+	if err != nil {
+		return nil
+	}
+	// get cross-regional resources
+	artifacts, err := opts.getArtifactBuckets()
+	if err != nil {
+		return nil
+	}
+
+	deployPipelineInput := &deploy.CreatePipelineInput{
+		ProjectName: opts.ProjectName(),
+		Name:        pipeline.Name,
+		Source: source,
+		Stages: stages,
+		ArtifactBuckets: artifacts,
+	}
+
+	// convert it to cloudfomration
+
+	// deploy it via CFN client
+	opts.prog.Start(fmt.Sprintf(fmtUpdatePipelineStart, color.HighlightUserInput(opts.PipelineFile)))
+
+	if err := opts.pipelineDeployer.DeployPipeline(deployPipelineInput); err != nil {
+		// check if pipeline already exists?
+		opts.prog.Stop(log.Serrorf(fmtUpdatePipelineFailed, color.HighlightUserInput(opts.PipelineFile)))
+		return fmt.Errorf("deploy pipeline: %w", err)
+	}
+	// TODO stream events
+
+	opts.prog.Stop(log.Ssuccessf(fmtUpdatePipelineComplete, color.HighlightUserInput(opts.PipelineFile))) // change to pipeline name?
+
+	return nil
+}
+
+// BuildPipelineUpdateCmd build the command for creating a new pipeline.
+func BuildPipelineUpdateCmd() *cobra.Command {
+	opts := NewUpdatePipelineOpts()
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Deploys a pipeline for applications in your workspace.",
+		Long:  `Deploys a pipeline for the applications in your workspace, using the environments associated with the applications.`,
+		Example: `
+  Deploy an updated pipeline for the applications in your workspace:
+  /code $ archer pipeline update`,
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			store, err := store.New()
+			if err != nil {
+				return fmt.Errorf("couldn't connect to project datastore: %w", err)
+			}
+			project, err := store.GetProject(opts.ProjectName())
+			if err != nil {
+				return fmt.Errorf("get project %s: %w", opts.ProjectName(), err)
+			}
+			opts.project = project
+
+			defaultSession, err := session.Default()
+			if err != nil {
+				return err
+			}
+			opts.pipelineDeployer = cloudformation.New(defaultSession)
+			region := aws.StringValue(defaultSession.Config.Region)
+
+			err = opts.pipelineDeployer.AddPipelineResourcesToProject(project, region)
```

Comment:
This could take up to two minutes, so we should probably move it to the execute block and show a spinner.

Reply:
Good point! Will fix